src/test/java/uk/nhs/hee/trainee/details/api/TraineeProfileResourceTest.java

```diff
@@ -254,13 +241,13 @@ void getShouldReturnBadRequestWhenPayloadNotMap() throws Exception {
   }
 
   @Test
-  void getShouldReturnNotFoundWhenTisIdNotInToken() throws Exception {
+  void getShouldReturnBadRequestWhenTisIdNotInToken() throws Exception {
     String token = TestJwtUtil.generateToken("{}");
 
     this.mockMvc.perform(get("/api/trainee-profile")
             .contentType(MediaType.APPLICATION_JSON)
             .header(HttpHeaders.AUTHORIZATION, token))
-        .andExpect(status().isNotFound());
+        .andExpect(status().isBadRequest());
```

Comment:
I suppose the new response for this and other requests without the token is not enough to comprise a breaking change.

Reply:
I don't believe so no, one of the benefits of this not being a document API (yet) 😉 

The previous `404` feels like an oversight anyway, we only get it because we previously passed `traineeId = null` from API -> service -> repo. Which is just a waste of processing.

At some point it will change again to return `403 Forbidden` when there is no trainee ID, which is what the new CCT endpoints will return. But the external (trainee facing) and internal (sync facing) endpoints need untangling first to make that easier to achieve (likely by replacing the internal calls with event listeners).